syntax = "proto3";
package CoSiMa-rpc;

service BaseInterface {
	rpc GetIntValue(string) returns int32 {}
	rpc GetBoolValue(string) returns bool {}
	rpc GetFloatValue(string) returns float {}
	rpc GetDoubleValue(string) returns double {}
	rpc GetStringValue(string) returns string {}

	rpc SetIntValue(NamedInt) returns int32 {}
	rpc SetBoolValue(NamedBool) returns int32 {}
	rpc SetFloatValue(NamedFloat) returns int32 {}
	rpc SetDoubleValue(NamedDouble) returns int32 {}
	rpc SetStringValue(NamedString) returns int32 {}
}

message NamedInt {
	string name = 1;
	int32 value = 2;
}

message NamedBool {
	string name = 1;
	bool value = 2;
}

message NamedFloat {
	string name = 1;
	float value = 2;
}

message NamedDouble {
	string name = 1;
	double value = 2;
}

message NamedString {
	string name = 1;
	string value = 2;
}